openlibrary/templates/type/edition/view.html

```diff
@@ -198,7 +198,7 @@
         $ edit_url = page.key + "?m=edit"
     </div>
       $:render_template("type/edition/compact_title", book_title, edit_url)
-      $:macros.EditionNavBar(work.edition_count, show_observations)
+      $:macros.EditionNavBar(get_observation_metrics(work['key']), work.edition_count, show_observations)
```

Comment:
```suggestion
      $ component_times['get_observation_metrics'] = time()
      $ reader_observations = get_observation_metrics(work['key']) if show_observations else {}
      $ component_times['get_observation_metrics'] = time() - component_times['get_observation_metrics']
      $:macros.EditionNavBar(reader_observations, work.edition_count, show_observations)
```
`get_observation_metrics` should only be called if `show_observations` is true.  Also, `get_observation_metrics` queries the database each time it is called.  Let's only call it one time per book page.  You can set `reader_observations` here, and remove these lines:
https://github.com/internetarchive/openlibrary/blob/76105d2009237a219953e0b96e38fc61d91d5d1e/openlibrary/templates/type/edition/view.html#L463-L466

__Note:__ `component_times` are used to show us how long different book page elements take to render.  You can view this information by adding a `debug=true` query parameter to any book page URL.

Reply:
Cheers for the feedback @jimchamp, very helpful! Will commit these suggestions now